examples/DataApi-NFT/template/modules/shared_data/src/main/scala/com/my/nft_example/shared_data/calculated_state/CalculatedState.scala

```diff
@@ -0,0 +1,49 @@
+package com.my.nft_example.shared_data.calculated_state
+
+import cats.effect.IO
+import com.my.nft_example.shared_data.types.Types.NFTUpdatesCalculatedState
+import eu.timepit.refined.types.numeric.NonNegLong
+import io.circe.syntax.EncoderOps
+import org.tessellation.schema.SnapshotOrdinal
+import org.tessellation.security.hash.Hash
+
+import java.security.MessageDigest
+
+object CalculatedState {
+
+  private var maybeNFTUpdatesCalculatedState: (SnapshotOrdinal, NFTUpdatesCalculatedState) = (
+    SnapshotOrdinal(NonNegLong(0L)),
+    NFTUpdatesCalculatedState(Map.empty)
+  )
+
+  def getCalculatedState: IO[(SnapshotOrdinal, NFTUpdatesCalculatedState)] = {
+    IO(maybeNFTUpdatesCalculatedState)
+  }
+
+  def setCalculatedState(snapshotOrdinal: SnapshotOrdinal, state: NFTUpdatesCalculatedState): IO[Boolean] = {
```

Comment:
It would be preferable to move `IO` out of this layer of that application but if we keep it here, I think we should at least wrap the entire computation in the `IO` with `IO.delay { ... }`